docs/appendices/release-notes/unreleased.rst

```diff
@@ -160,6 +160,9 @@ Fixes
 .. stable branch. You can add a version label (`v/X.Y`) to the pull request for
 .. an automated mergify backport.
 
+- Optimized the evaluation of ``CASE`` expressions to prevent stack overflows
```

Comment:
Shouldn't this be a change? it's not planned to be backported AFAIK.

Reply:
Maybe it should be phrased differently.